### ISSUE TITLE
Support `valid/` folder as validation set in classification task ("val" and "validation" are already options)

### ISSUE THREAD

**Issue by JamesBond6873** [ultralytics/ultralytics#21311]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

When exporting a classification dataset from Roboflow, the folder structure created is:
train/
valid/
test/

However, the current Ultralytics code for classification datasets only checks for `val/` and `validation/` as the validation folder:

```
val_set = (
    data_dir / "val"
    if (data_dir / "val").exists()
    else data_dir / "validation"
    if (data_dir / "validation").exists()
    else None
)

```
Since valid/ does not match either val/ or validation/, the trainer skips it and instead uses test/ as the fallback. This happens even though valid/ is present and is the intended validation set, and Roboflow’s export structure seems to be quite common among users.

Proposed Change:
Expand the fallback logic to also support valid/, for example:

```
val_set = (
    data_dir / "val"
    if (data_dir / "val").exists()
    else data_dir / "validation"
    if (data_dir / "validation").exists()
    else data_dir / "valid"
    if (data_dir / "valid").exists()
    else None
)
```

Would this be a worthwhile change to include in the main repository? If so, I am happy to open a pull request implementing this small improvement.

### Use case

This feature would make the repo more compatible with Roboflow naming conventions for classification datasets exports. More specifically, it would work with Roboflow naming convention "valid" for the validation dataset.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JamesBond6873, thank you for your suggestion and for your interest in improving Ultralytics 🚀! This is an automated response to let you know we've received your feature request. An Ultralytics engineer will review your proposal and assist you soon.

For more information on using Ultralytics, please visit our [Docs](https://docs.ultralytics.com/) where you'll find helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

If you're working on a 🐛 Bug Report, please make sure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven't already, as this greatly helps us understand and address the issue.

If you have questions about custom training or datasets, please include as much detail as possible, such as dataset examples and training logs, and check our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time discussions, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For longer-form questions, try [Discourse](https://community.ultralytics.com/). You can also join conversations on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to connect with others.

## Upgrade

Before submitting a pull request or making further changes, please ensure you're using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following verified environments, each with up-to-date dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. Our CI validates YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your contribution and willingness to help! An Ultralytics engineer will follow up with you shortly.

**Comment by glenn-jocher** [comment]
Thanks for the detailed feature request! This is a reasonable improvement that would enhance compatibility with Roboflow exports. You've correctly identified the relevant code in [`ultralytics/data/utils.py`](https://docs.ultralytics.com/reference/data/utils/) in the `check_cls_dataset` function.

Your proposed change looks good - adding `valid/` as another fallback option would be helpful for users working with Roboflow datasets. Please go ahead and submit a PR with this enhancement, and make sure to include a brief test case if possible to verify the new functionality works as expected.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in PR #21321, which improves classification dataset compatibility by allowing "valid" to be recognized as a fallback validation folder, in addition to "val" and "validation". The update also brings clearer type hints and some code cleanups for better readability and maintainability. Thanks to the efforts of the PR author and contributors for enhancing the dataset utility functions!

To test if this fix resolves the issue, try installing the latest code from the main branch:

```
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Alternatively, you can wait for the next official release on PyPI, where these changes will be included.

Please let us know if the issue is resolved after updating, or if you continue to experience any problems. Thanks again for reporting this and helping to improve Ultralytics!
